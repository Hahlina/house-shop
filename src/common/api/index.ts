export * from './fetchApi';
export * from './services/propety';
