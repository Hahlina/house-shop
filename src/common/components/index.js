export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
