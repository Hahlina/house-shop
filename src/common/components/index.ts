export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
export * from './Container/Container';
export * from './Image/Image';
export * from './ErrorPage/ErrorPage';
export * from './SomethingWentWrong/SomethingWentWrong';
export * from './NothingFound/NothingFound';
