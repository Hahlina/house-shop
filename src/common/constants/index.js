export * from './config';
export * from './apiRoutes';
export * from './initValues';
export * from './appRoutes';
