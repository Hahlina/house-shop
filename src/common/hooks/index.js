export * from './useProperties';
export * from './useProperty';
