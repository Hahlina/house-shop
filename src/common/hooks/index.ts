export * from './state';
export * from './useImageOnLoad';
