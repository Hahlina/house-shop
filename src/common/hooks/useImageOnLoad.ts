import { type CSSProperties, useState } from 'react';

interface ImageStyle {
    thumbnail: CSSProperties;
    fullSize: CSSProperties;
}

interface ImageOnLoadType {
    handleImageOnLoad: () => void;
    css: ImageStyle;
    isLoaded: boolean;
}

export const useImageOnLoad = (): ImageOnLoadType => {
    const [isLoaded, setIsLoaded] = useState<boolean>(false);

    const handleImageOnLoad = () => {
        setIsLoaded(true);
    };

    const css: ImageStyle = {
        thumbnail: {
            visibility: isLoaded ? 'hidden' : 'visible',
            transition: 'visibility 200ms ease-out 700ms'
        },
        fullSize: {
            opacity: isLoaded ? 1 : 0,
            transition: 'opacity 800ms ease-in 0ms'
        }
    };

    return { isLoaded, handleImageOnLoad, css };
};
