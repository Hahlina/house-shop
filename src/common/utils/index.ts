export * from './convertPrice';
export * from './capitalizeFirstLetter';
