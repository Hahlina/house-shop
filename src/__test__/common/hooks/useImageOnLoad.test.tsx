import { useImageOnLoad } from 'common/hooks';
import { renderHook, act } from 'common/utils/test';

describe('useImageOnLoad', () => {
    it('should set isLoaded to true after handling image onLoad', () => {
        const { result } = renderHook(() => useImageOnLoad());
        act(() => {
            result.current.handleImageOnLoad();
        });
        expect(result.current.isLoaded).toBe(true);
    });

    it('should set thumbnail visibility to hidden after handling image onLoad', () => {
        const { result } = renderHook(() => useImageOnLoad());
        act(() => {
            result.current.handleImageOnLoad();
        });
        expect(result.current.css.thumbnail.visibility).toBe('hidden');
    });

    it('should set fullSize opacity to 1 after handling image onLoad', () => {
        const { result } = renderHook(() => useImageOnLoad());
        act(() => {
            result.current.handleImageOnLoad();
        });
        expect(result.current.css.fullSize.opacity).toBe(1);
    });

    it('should set thumbnail visibility to visible before handling image onLoad', () => {
        const { result } = renderHook(() => useImageOnLoad());
        expect(result.current.css.thumbnail.visibility).toBe('visible');
    });

    it('should set fullSize opacity to 0 before handling image onLoad', () => {
        const { result } = renderHook(() => useImageOnLoad());
        expect(result.current.css.fullSize.opacity).toBe(0);
    });

    it('should take a snapshot', () => {
        const { result } = renderHook(() => useImageOnLoad());
        expect(result).toMatchSnapshot();
    });
});
