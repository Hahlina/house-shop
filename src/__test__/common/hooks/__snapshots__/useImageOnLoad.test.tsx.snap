// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useImageOnLoad should take a snapshot 1`] = `
Object {
  "current": Object {
    "css": Object {
      "fullSize": Object {
        "opacity": 0,
        "transition": "opacity 800ms ease-in 0ms",
      },
      "thumbnail": Object {
        "transition": "visibility 200ms ease-out 700ms",
        "visibility": "visible",
      },
    },
    "handleImageOnLoad": [Function],
    "isLoaded": false,
  },
}
`;
